cmd/osbuild-worker/jobimpl-osbuild.go

```diff
@@ -30,14 +30,19 @@ import (
 )
 
 type OSBuildJobImpl struct {
-	Store          string
-	Output         string
-	KojiServers    map[string]koji.GSSAPICredentials
-	GCPCreds       string
-	AzureCreds     *azure.Credentials
-	AWSCreds       string
-	AWSBucket      string
-	GenericS3Creds string
+	Store                        string
+	Output                       string
+	KojiServers                  map[string]koji.GSSAPICredentials
+	GCPCreds                     string
+	AzureCreds                   *azure.Credentials
+	AWSCreds                     string
+	AWSBucket                    string
+	GenericS3Creds               string
```

Comment:
I did not want to unify the details of the AWS and Generic S3 because the AWS details are used not only when uploading to S3 but also when building an AWS AMI (the image is uploaded to the same S3 bucket and then imported as an AMI).

As for the struct, should I create a struct just for the Generic S3 case (with `Creds`, `Endpoint`, `Region`, `Bucket`, `CABundle` and `SSLVerification`)? Or also for the AWS details (same struct, but with only the required details)

Reply:
IMO long term the aws details should become ec2 specific config (not in the PR), and generic s3 config should be just for s3. But to make sure we don't break it currently, we need to fill the bucket with the AWSBucket (ec2 config) details.

Reply:
That's a valid point, @Gundersanne. Anyway, a struct for (generic) S3 seems like a good idea to me.

Reply:
Yea we can still have the separate struct and then fill it for now until we have all the configs changed everywhere.

Reply:
I've put all S3 configuration values into a new struct inside the `OSBuildJobImpl` struct